src/otx/engine/engine.py

```diff
@@ -533,7 +533,7 @@ def optimize(
 
         model = self.model
         if not isinstance(model, OVModel):
-            datamodule = self._auto_configurator.get_ov_datamodule()
+            optimize_datamodule = self._auto_configurator.update_ov_test_pipeline(datamodule=optimize_datamodule)
```

Comment:
It is basically not allowed to use a test set for model optimization. Anyway, why do we have to create a new datamodule even if we already has `self.datamodule`?

Reply:
This is the code added by https://github.com/openvinotoolkit/training_extensions/pull/3028 in that PR. 
@sovrasov Could you please answer this question?

Reply:
I removed the syntax to update the datamodule in optimize. - https://github.com/openvinotoolkit/training_extensions/pull/3139/commits/edadbb6434ecfe51b0d17c3c901f098b09817589

Reply:
> Anyway, why do we have to create a new datamodule even if we already has self.datamodule?

This is not just for the optimize. Why do we need this recreation?

Reply:
I don't know the details, but there was a request from Vlad to update the test pipeline use the pipeline in `openvino_model.yaml` when using `OVModel`.

Reply:
I see the reason. Can you update this to update the train pipeline with the OV data pipeline such as

```python
# not update_ov_test_pipeline()
optimize_datamodule = self._auto_configurator.update_ov_train_pipeline(datamodule=optimize_datamodule)
```

Reply:
I modified the `update_ov_test_pipeline` function to `update_ov_subset_pipeline` to allow it to be updated by specifying a subset. - https://github.com/openvinotoolkit/training_extensions/pull/3139/commits/13ac06895c230a7c02312ae757d62b95b68c906b